src/OpenTelemetry.Exporter.Console/ConsoleExporter.cs

```diff
@@ -34,116 +29,13 @@ public ConsoleExporter(ConsoleExporterOptions options)
-                            foreach (var attribute in activityEvent.Tags)
-                            {
-                                this.WriteLine($"        {attribute.Key}: {attribute.Value}");
-                            }
-                        }
-                    }
-
-                    if (activity.Baggage.Any())
-                    {
-                        this.WriteLine("Activity.Baggage:");
-                        foreach (var baggage in activity.Baggage)
-                        {
-                            this.WriteLine($"    {baggage.Key}: {baggage.Value}");
-                        }
-                    }
-
-                    var resource = this.ParentProvider.GetResource();
-                    if (resource != Resource.Empty)
-                    {
-                        this.WriteLine("Resource associated with Activity:");
-                        foreach (var resourceAttribute in resource.Attributes)
-                        {
-                            this.WriteLine($"    {resourceAttribute.Key}: {resourceAttribute.Value}");
-                        }
-                    }
-
-                    this.WriteLine(string.Empty);
-                }
-            }
-#if NET461 || NETSTANDARD2_0
-            else if (typeof(T) == typeof(LogRecord))
-            {
-                var rightPaddingLength = 30;
-                foreach (var item in batch)
-                {
-                    var logRecord = item as LogRecord;
-                    this.WriteLine($"{"LogRecord.TraceId:".PadRight(rightPaddingLength)}{logRecord.TraceId}");
-                    this.WriteLine($"{"LogRecord.SpanId:".PadRight(rightPaddingLength)}{logRecord.SpanId}");
-                    this.WriteLine($"{"LogRecord.Timestamp:".PadRight(rightPaddingLength)}{logRecord.Timestamp:yyyy-MM-ddTHH:mm:ss.fffffffZ}");
-                    this.WriteLine($"{"LogRecord.EventId:".PadRight(rightPaddingLength)}{logRecord.EventId}");
-                    this.WriteLine($"{"LogRecord.CategoryName:".PadRight(rightPaddingLength)}{logRecord.CategoryName}");
-                    this.WriteLine($"{"LogRecord.LogLevel:".PadRight(rightPaddingLength)}{logRecord.LogLevel}");
-                    this.WriteLine($"{"LogRecord.TraceFlags:".PadRight(rightPaddingLength)}{logRecord.TraceFlags}");
-                    this.WriteLine($"{"LogRecord.State:".PadRight(rightPaddingLength)}{logRecord.State}");
-                    if (logRecord.Exception is { })
-                    {
-                        this.WriteLine($"{"LogRecord.Exception:".PadRight(rightPaddingLength)}{logRecord.Exception?.Message}");
-                    }
-
-                    this.WriteLine(string.Empty);
-                }
-            }
-#endif
-
-            return ExportResult.Success;
-        }
+        public abstract override ExportResult Export(in Batch<T> batch);
 
-        private void WriteLine(string message)
+        internal void WriteLine(string message)
```

Comment:
You don't like `protected` for this? The benefit to that is `ConsoleExporter<T>` is public so consumers (potentially) making their own derived types could benefit from the `WriteLine` functionality. Alternatively, if we want to keep this `internal`, maybe we should make `ConsoleExporter<T>` ctor `internal` to prevent anyone from making their own derived types?

/cc @cijothomas

Reply:
I think we should make the ConsoleExporter<T> ctor internal as it wouldn't make sense to have ConsoleExporters for anything other than Activity, LogRecord, and Metrics. If we make this internal we would have to make ConsoleActivityExporter and ConsoleLogRecordExporter internal as well. We would only allow adding ConsoleExporter through the public extension method then.

Reply:
I'm good with that.

Reply:
We may want to keep ConsoleActivityExporter/ConsoleLogRecordExporter public. (And Zipkin/Jaeger/Otlp as well). This is required if we want to allow users to write FilterProcessing themselves.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/1559

Reply:
I have updated the access modifier to protected.